### DISCUSSION TITLE
tailwindcss latest version and wordpress theme --  remove "JIT"mode, can compile style.css but can't generate full style code.

### DISCUSSION THREAD

**Discussion by awsliu** [tailwindlabs/tailwindcss#12414] [Help]
All content in WordPress is stored in the pages module. By editing the content of pages in the WordPress backend, all data is saved in the database. Therefore, when using Tailwind CSS, we also save some class names in the pages.

I have been using Tailwind CSS in my WordPress theme, previously with version 1.9.6, and it worked well. Running npm run dev could compile and execute the code. Although the generated CSS file was large, it successfully generated all styles related to screen size changes. Our configuration defined in tailwind.config.js is as follows:

module.exports = {
  theme: {
    screens: {
      'xs': {'max': '600px'},
      'sm': '601px',
      'smx': {'max': '849px'},
      'md': '850px',
      'mdx': {'max': '1023px'},
      'lg': '1024px',
      'lgx': {'max': '1279px'},
      'xl': '1280px',
      'xlx': {'max': '1535px'},
      '2xl': '1536px',
    },
    extend: {
    },
  }
}

However, after upgrading Tailwind CSS to the latest version 3.3.5, and running npm install -D tailwindcss@latest postcss@latest autoprefixer@latest, and npm install for upgrade all node modules and making modifications according to the Tailwind CSS documentation, although it compiles successfully and execute small size style file. But we lost all screen styles from style.css. These changes cause our wordpress page style became chaotic after the upgrade. According to the Tailwind CSS documentation, hiding or commenting out the JIT settings should generate the originally large-sized style files. However, despite our repeated tests, we have not been able to achieve this. How should we configure it so that we can still export large-sized style files while using the latest version of Tailwind CSS?

**Comment by wongjn** [comment]
In Tailwind v3, there was a paradigm shift whereby the only mode is now JIT. You'd want to cover your template files in your theme, and then perhaps use a similar method to [`@tailwindcss/typography`](https://tailwindcss.com/docs/typography-plugin) to style content from a database.

Otherwise, to have similar behavior to non-JIT, you'd want to look at configuring a broad [`safelist`](https://tailwindcss.com/docs/content-configuration#safelisting-classes) like:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  safelist: [
    {
      pattern: /^/,
      variants: [
        'xs',
        'sm',
        // …
      ],
    },
  ],
  // …
}
```

**Reply by awsliu** [reply]
Hi [wongjn], thanks your reply,

But i try again with your way and it's not working. 

I attached the file structures and webpack.mix.js, tailwind.config.js, and package.json content. Please see if there is any problem on our configuration files. Thanks a lot,

<img width="537" alt="Screenshot 2023-11-20 at 12 38 15 pm" src="https://github.com/tailwindlabs/tailwindcss/assets/150862960/33f96850-9100-4d80-adad-05d047958941">
<img width="416" alt="Screenshot 2023-11-20 at 12 38 21 pm" src="https://github.com/tailwindlabs/tailwindcss/assets/150862960/1b331c49-73bc-4fbc-97f6-32cabe1bc088">
<img width="1339" alt="Screenshot 2023-11-20 at 12 38 55 pm" src="https://github.com/tailwindlabs/tailwindcss/assets/150862960/71c82a65-5d7a-44a9-ad95-c68fab57e22a">
<img width="216" alt="Screenshot 2023-11-20 at 12 40 55 pm" src="https://github.com/tailwindlabs/tailwindcss/assets/150862960/122969cb-c5c2-48cc-86e4-c184e0cc4673">

**Reply by shreshthmohan** [reply]
You need to add all your "screen styles" to the safelist. So if you have any styles in, say, the blog content, those need to be added to the safelist.

But as @wongjn suggested above, use [@tailwindcss/typography](https://tailwindcss.com/docs/typography-plugin) with some [overrides](https://tailwindcss.com/docs/typography-plugin#customizing-the-css) instead. If you use this approach, you might not need a safelist at all.

Why not use safelist:

> Safelisting is a last-resort, and should only be used in situations where it’s impossible to scan certain content for class names. These situations are rare, and you should almost never need this feature.

From Tailwind CSS docs on [safelisting](https://tailwindcss.com/docs/content-configuration#safelisting-classes)

**Comment by awsliu** [comment]
@wongjn @shreshthmohan Thanks your help, finally i got a solution with [@tailwindcss/typography](https://tailwindcss.com/docs/typography-plugin)

1. Install typography with "npm install -D @tailwindcss/typography"
2. Setup postcss.config.js
3. Setup a temporary html file, e.g test.html, add all class name and screen class name to html file. e.g <div class="text-2xl md:text-3xl">Test</div>
4. Run "npm run dev" and compile the output css file.

Using the above method, whenever we add a new class name, we need to perform a compilation. Is there any way to add all tailwindcss class names in the output.css file during the development phase? And when it comes to the product stage, only the class names involved will be compiled?
